Terraform/Home_Task_2/modules/asg/main.tf

```diff
@@ -42,73 +43,52 @@ resource "aws_autoscaling_group" "this_asg" {
   max_size             = 4
   desired_capacity     = 2
   vpc_zone_identifier  = [var.private_subnet_1_id, var.private_subnet_2_id]
-  #availability_zones = ["us-east-1a"]
-  target_group_arns = [aws_alb_target_group.this_tg.arn]
+}
 
-  lifecycle {
-    ignore_changes = [load_balancers, target_group_arns] #[aws_lb.alb.arn, aws_alb_target_group.this_tg.arn]
-  }
-  depends_on = [aws_lb.alb]
+resource "aws_autoscaling_attachment" "asg_attachment_bar" {
+  autoscaling_group_name = aws_autoscaling_group.this_asg.id
+  lb_target_group_arn    = var.tg_arn
 }
 
-#-------------------------------------application load balancer---------------------------------------
-resource "aws_lb" "alb" {
-  name               = "lb-terraform"
-  internal           = false
-  load_balancer_type = "application"
-  ip_address_type    = "ipv4"
-  security_groups    = [var.alb_sg]
-  subnets            = [var.public_subnet_1_id, var.public_subnet_2_id]
+resource "aws_security_group" "sg_public" {
+  name   = "sg_pub_1"
+  vpc_id = var.vpc_id
 
-  enable_deletion_protection = false
-  /*
-  access_logs {
-    bucket  = aws_s3_bucket.lb_logs.bucket
-    prefix  = "test-lb"
-    enabled = true
+  ingress {
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+    from_port   = 22
+    to_port     = 22
   }
-*/
-  tags = {
-    Environment = "production"
-    Name        = "alb"
+  egress {
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+    from_port   = 0
+    to_port     = 0
   }
 }
 
-resource "aws_alb_target_group" "this_tg" {
-  name        = "terraform-alb-target"
-  port        = 80
-  protocol    = "HTTP"
-  vpc_id      = var.vpc_id
-  target_type = "instance"
-  stickiness {
-    type = "lb_cookie"
+resource "aws_security_group" "sg_private" {
+  name   = "sg_priv_1"
+  vpc_id = var.vpc_id
+
+  ingress {
+    protocol = "tcp"
+    #cidr_blocks = ["0.0.0.0/0"] #["10.0.1.0/24"]
+    security_groups = [aws_security_group.sg_public.id]
+    from_port       = 80

```

Comment:
use for from_port and to_port params only 22 port (as a variable from general variables.tf file)
I've written about this before!!